### PR TITLE
Add Filters

### PR DESCRIPTION
**Pull request by gadgetfather**
## Files to check 

- filter-context.js
- ProductGrid.jsx
- Aside.jsx
- ProductCard.jsx

# Preview
ezgif com-gif-maker

### CODE REVIEW

**Review by dipteekhd — APPROVED** [review]
In filter-context.js
1.You can extract filterReducer function in separate file and import
2. use arrow function for filterReducer function

**Comment by AnkurChunekar** [comment]
Hey Akshay if possible provide a csb link, it helps to review functionalities easily

**Comment by AnkurChunekar** [comment]
akshay pr
please check the contrast ratio for min and max price values

**Comment by AnkurChunekar** [comment]
akshay pr 2
- Check this when clicked on RPG category it is showing sports and when on sports it show rpg please correct this one

akshay 3
- If i click on all the 3 or any 2 no items are listed please look into this.

- add id and for attribute to the filters for better accessibility

File: src/components/Aside/Aside.jsx [hunk 1]
```diff
@@ -1,69 +1,151 @@
 import React from "react";
+import { useFilter } from "../../context/filter-context";
 import "./Aside.css";
+
 export function Aside() {
```
Comment (AtulPant2704) [inline src/components/Aside/Aside.jsx:5]: You can rename this as Filter.

File: src/components/Aside/Aside.jsx [hunk 2]
```diff
@@ -1,69 +1,151 @@
...
+          />
           <label htmlFor="LtoH">Low to High</label>
         </span>
-        <span>
-          <input type="radio" name="sort" />
-          <label htmlFor="NtoO">New to Old</label>
-        </span>
-        <span>
-          <input type="radio" name="sort" />
-          <label htmlFor="OtoN">Old to New</label>
```
Comment (AnkurChunekar) [inline src/components/Aside/Aside.jsx:64]: IMO you can improve on namings here for id and for attribute.

File: src/components/ProductCard/ProductCard.css [hunk 3]
```diff
@@ -4,6 +4,7 @@
   margin-bottom: 0.8rem;
   margin-right: 9.75px;
   margin-left: 9.75px;
```
Comment (AnkurChunekar) [inline src/components/ProductCard/ProductCard.css:6]: you can use 8px for margin bottom also just for consistency

File: src/components/ProductCard/ProductCard.css [hunk 4]
```diff
@@ -14,12 +15,24 @@
...
+
+.img-container {
+  height: 200px;
+  position: relative;
+}
+.img-container .card-tag {
+  background-color: black;
+  position: absolute;
+  width: fit-content;
+  padding: 0.2rem 0.7rem;
```
Comment (AnkurChunekar) [inline src/components/ProductCard/ProductCard.css:31]: here also

File: src/context/filter-context.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        return { ...state, rating: action.payload };
+      case "SLIDER":
+        return { ...state, sliderValue: action.payload };
+      case "ACTION":
+        return { ...state, categoryAction: !state.categoryAction };
+      case "SPORTS":
+        return { ...state, categorySports: !state.categorySports };
+      case "RPG":
+        return { ...state, categoryRPG: !state.categoryRPG };
+      default:
```
Comment (AnkurChunekar) [inline src/context/filter-context.js]: great work here this looks very clean.

**Review by AtulPant2704 — COMMENTED** [review]
- As Diptee mentioned you can extract out filterReducer in a separate file.
- You can extract out all the filter and sort functions in separate files out of the productListing page.
- You can also extract out each of the filter like filter for rating and filter by category as a separate component.
Following all the above steps, your code would look much clean and will help you in debugging later on.
- And what Ankur mention that on clicking of more than one category, your products are not getting displayed, so work on that functionality first.

File: src/components/Aside/Aside.jsx [hunk 6]
```diff
@@ -1,138 +1,313 @@
...
+        </div>
         <h2>Category</h2>
         <span>
-          <input type="checkbox" />
+          <input
+            type="checkbox"
+            checked={categoryAction}
+            onChange={() => filterDispatch({ type: "ACTION" })}
+          />
           <label htmlFor="Action">Action</label>
```
Comment (AnkurChunekar) [inline src/components/Aside/Aside.jsx:57]: - you have included for attribute for all label tags, also add id to the all of the inputs for better accessibility.
Rest everything looks great, Nice work akshay.

**Review by AnkurChunekar — APPROVED** [review]

**Pull request merged**